packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -145,6 +145,15 @@ function plainFunction(path: NodePath, callId: Object) {
     ]);
     declar._blockHoist = true;
 
+    if (path.parentPath.isExportDefaultDeclaration()) {
+      // change the path type so that replaceWith() does not wrap
+      // the identifier into an expressionStatement
+      path.type = "FunctionExpression";
+      path.parentPath.insertBefore(declar);
+      path.replaceWith(t.identifier(asyncFnId.name));
```

Comment:
Yeah that was the motivation for `This isn't quite correct`. In the future I'd love to just have this transform not convert things to expressions at all in favor of breaking it up instead or something, like

```
export default async function fn(){
  await 4;
}
```
becoming

```
function * fnImpl(){
  yield 4;
}
var impl;
export default function fn(){
  impl = impl || babelHelpers.asyncToGenerator(fnImpl);

  return impl.apply(this, arguments);
}
```
Not sure what performance on that would be like but I'd guess no worse than now?